### ISSUE TITLE
Please consider of next official release

### ISSUE THREAD

**Issue by inductor** [h2o/h2o#2941]
It's been more than two years since the last release (2.2.6) and 2.3.0 is still not released officially.

It is a big problem especially for server admins for those who want to use official release with latest features

**Comment by mrdomino** [comment]
Concurred — in particular it would be nice to make use of a tagged release with the stats tracking code in it.

**Comment by LoganDark** [comment]
> It is a big problem especially for server admins who want to use official release with latest features

You can just install the latest version for your distro, then build it locally and manually replace the `h2o`, `start_server` and `ca-bundle.crt` files. That's what I did for http3 support

**Comment by inductor** [comment]
@LoganDark I don't think you'd like to that for each host when you have bunch of instances(and I'm not talking about creating a baked image like using Packer)

**Comment by LoganDark** [comment]
> I don't think you'd like to that for each host when you have bunch of instances

If you have a fleet of servers you should know how to deploy a couple extra files to all of them without too much effort. It's literally 3 files to replace. Run a shell script that does the replacement.

**Comment by omitter** [comment]
I think you should absolutely invest some time into a new release! The current state makes this project **feel** like abandonware. And impression is key for adoption. I think you're sabotaging your own success like this. :disappointed:

Let me explain a bit:
I tried to give h2o a chance and replace Nginx by it, mainly because of its speed promises, easier configuration, and mruby flexibility. But you make it difficult to commit on using it.

This is what hit me:
- The changes file was last updated 5 years ago → First thought: it's dead since 5 years.
- The benchmark listed on the website compare ancient versions → Is h2o still fast today?
- The blog linked on the website has its last entry in 2020 → Only maintenance, no active development maybe?
- I'm using the Ubuntu package (2.2) and the documentation is out of sync with that version (stash and strict-match didn't work for me) → How are package maintainers supposed to release a current version of your software if you don't tag?
- When checking pulse and commit insights, I can still see some action, but no release → Looks like something is wrong with the project.
- A lot of open old issues → Are they still relevant? Nobody seems to care about bookkeeping (I don't blame you).
- According to this [blog](https://eklausmeier.goip.de/blog/2024/04-13-installing-and-configuring-the-h2o-web-server) mruby version is too old by now.

If you really think it is too hard to group features into version numbers, maybe just make periodical releases? Just use the release date as version number, something like `h2o-2024-08` and make a new release every half or full year and between for security issues.

**Comment by volyrique** [comment]
Are you aware of #3230? IMHO this issue is outdated.

**Comment by omitter** [comment]
Yes, I am. And I think the switch introduced in #3230 is a big mistake.
How are package maintainers supposed to release a the current version of a software if it has no versions?

**Comment by volyrique** [comment]
The commit IDs are the versions. In any case, if you disagree with the new release model, the other issue is a more appropriate place to say that.

**Comment by fetus-hina** [comment]
Hi.
I'm creating unofficial (and not linked from this project) RPM packages.
I release the RPM packages with the following version number for consistency with the changes and previous versions: `h2o-2.3.0-1.git20241128.0.f7a0e1d6e.el9.jp3cki.x86_64.rpm`.
`2.3.0` no longer has any meaning, and the version is identified by the commit date (`20241128`) and commit ID (`f7a0e1d6e`).

Not only for this project, it is not uncommon for the officially released version to be too old to be of any practical use.
In such cases, it is necessary to make a package of the `HEAD` commit, a specific branch or a specific revision (commit).

The problem for many package maintainers is that it is difficult to know when to update.
The package maintainers can always release the newest (`HEAD`), or they can choose to update once a month, or they can respond only when Kazuho releases vulnerability information.

**Comment by omitter** [comment]
I'm aware that it is not impossible to create packages, I just say it makes it unnecessarily hard to do so. It's hard for you as a packager (what changed, when do you need a new update) and for the users too (which *version* supports which features, it is difficult to read the docs). It also gives a *feeling* that the devs are not really interested in the project, if they don't want to publish a new version.
Btw, for the same reason every project should have a change log (preferably with version and time info). I understand that it is annoying for the devs to fill it in, but it saves a lot of trouble for the users/packagers. Documentation really matters!

**Comment by pbowyer** [comment]
As a compromise, could h2o add a date-based tag at the end of every day when commits have been added to the `master` branch? I'd hope this could be automated so the maintainers have to do nothing.

**Comment by h4rldev** [comment]
h2o has moved to a new type of versioning scheme, meaning that any sort of push to main/master is considered production, if you want releases compile it or set up a compilation system that automatically compiles regularly and pushes out a new version. Not sure if Kazuho would appreciate it though.

**Comment by h4rldev** [comment]
#3230

**Comment by h4rldev** [comment]
tl;dr compile it urself or find releases someone else compiles themselves and just be careful cause stuff can break possibly.

**Comment by Dentrax** [comment]
Hey, it seems the last release was [Aug 2019](https://github.com/h2o/h2o/releases), and this is still the case. But main branch has been actively maintaining. Any news for versioning policy?
